libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2064,6 +2064,37 @@ bool WaveTrack::Get(samplePtr buffer, sampleFormat format,
    return result;
 }
 
+LibSampleTrack::TrackBlockSequence
+WaveTrack::GrabFloats(double& t0, sampleCount length) const
+{
+   WaveClipConstHolders intersectingClips;
+   const auto t1 = t0 + length.as_double() / mRate;
+   for (const auto& clip : mClips)
+      if (t0 < clip->GetPlayEndTime() && clip->GetPlayStartTime() < t1)
+         intersectingClips.push_back(clip);
+   if (intersectingClips.empty())
+      return {};
+   std::sort(
+      intersectingClips.begin(), intersectingClips.end(),
+      [](const auto& a, const auto& b) {
+         return a->GetPlayStartTime() < b->GetPlayStartTime();
+      });
+   std::vector<LibSampleTrack::AdjacentBlockSequence> blocks(intersectingClips.size());
+   std::vector<sampleCount> blockOffsets(intersectingClips.size());
+   for (auto i = 0u; i < intersectingClips.size();++i)
+   {
+      const auto& clip = intersectingClips[i];
+      const auto clipT0 = std::max(t0 - clip->GetPlayStartTime(), 0.);
+      const sampleCount start { clipT0 * clip->GetRate() };
+      const sampleCount len { (t1 - clipT0) * clip->GetRate() };
+      auto newBlocks = clip->GrabFloats(start, len);
+      t0 += newBlocks.usefulLength.as_double() / GetRate();
+      blocks[i] = clip->GrabFloats(start, len);
```

Comment:
Oj, one `clip->GrabFloats(start, len);` too many ; useless but doesn't break anything.

Reply:
Yes, why not instead
```
blocks[i] = std::move(newBlocks);
```